charts/k8ssandra/templates/_helpers.tpl

```diff
@@ -161,15 +161,54 @@ Set default num_tokens based on the server version
 */}}
 {{- define "k8ssandra.default_num_tokens" -}}
 {{- $datacenter := (index .Values.cassandra.datacenters 0) -}}
-    {{- if $datacenter.num_tokens }}
-      num_tokens: {{ $datacenter.num_tokens }}
-    {{- else }}
-    {{- if hasPrefix "3.11" .Values.cassandra.version }}
-      num_tokens: 256
-    {{- else }}
-      num_tokens: 16
-    {{- end }}
+{{- if .Release.IsInstall }}
+  {{- /*
+  If num_tokens is set we simply use that value.
+  */}}
+  {{- if $datacenter.num_tokens }}
```

Comment:
Won't this fail if user didn't set it originally? Yes, it's in the created CassandraDatacenter, but it's not in the Helm's internal properties.